src/main/java/org/jabref/logic/openoffice/action/Update.java

```diff
@@ -0,0 +1,133 @@
+package org.jabref.logic.openoffice.action;
+
+import java.util.List;
+
+import org.jabref.logic.openoffice.frontend.OOFrontend;
+import org.jabref.logic.openoffice.frontend.UpdateBibliography;
+import org.jabref.logic.openoffice.frontend.UpdateCitationMarkers;
+import org.jabref.logic.openoffice.style.OOBibStyle;
+import org.jabref.logic.openoffice.style.OOProcess;
+import org.jabref.model.database.BibDatabase;
+import org.jabref.model.openoffice.rangesort.FunctionalTextViewCursor;
+import org.jabref.model.openoffice.uno.CreationException;
+import org.jabref.model.openoffice.uno.NoDocumentException;
+import org.jabref.model.openoffice.uno.UnoScreenRefresh;
+
+import com.sun.star.lang.WrappedTargetException;
+import com.sun.star.text.XTextDocument;
+
+/**
+ * Update document: citation marks and bibliography
+ */
+public class Update {
+
+    private Update() {
+        /**/
+    }
+
+    /**
+     * @return the list of unresolved citation keys
+     */
+    private static List<String> updateDocument(XTextDocument doc,
+                                               OOFrontend frontend,
+                                               List<BibDatabase> databases,
+                                               OOBibStyle style,
+                                               FunctionalTextViewCursor fcursor,
+                                               boolean doUpdateBibliography,
+                                               boolean alwaysAddCitedOnPages)
+        throws
+        CreationException,
+        NoDocumentException,
+        WrappedTargetException,
+        com.sun.star.lang.IllegalArgumentException {
+
+        final boolean useLockControllers = true;
```

Comment:
If this is a debug constant, it should be
```suggestion
        static final boolean USE_LOCK_CONTROLLERS = true;
```

Reply:
ok